Life.java

```diff
@@ -1,55 +1,101 @@
 public class Life implements ILife {
+  int width;
+  int height;
+  int[][] board;
   
   public static void main(String[] args) {
-    Life l = new Life(new String[] {  "     ",
-                                      "     ",
-                                      " *** ",
-                                      "     ",
-                                      "     " });
-    l = (Life) l.nextGeneration();
   }
 
-
-  public Life() {
-    nukeAll();
+  public Life(int width, int height) {
+    this.width = width;
+    this.height = height;
+    this.board = new int[width][height];
   }
 
-  public Life(String[] setup) {
-    this();
-    for (int y = 0; y < setup.length; y++)
-      for (int x = 0; x < setup[y].length(); x++)
-        if (setup[y].charAt(x) != ' ')
-          setAlive(x, y);
+  // Just for testing the Graphical output of the Game of Life
+  public void printBoard(){
+    for(int y = 0; y < height; y++){
+      String line = "|";
+      for(int x = 0; x < width; x++){
+        if(this.board[x][y] == 0){
+          line += ".";
+        }else{
+          line += "*";
+        }
+      }
+      line += "|";
+      System.out.println(line);
+    }
+    System.out.println("\n");
   }
 
 
   @Override
   public void nukeAll() {
-    // TODO Auto-generated method stub
+    for(int y = 0; y < height; y++){
+      for(int x = 0; x < width; x++){
+        this.board[x][y] = 0;
+      }
+    }
 
   }
 
   @Override
   public void setAlive(int x, int y) {
-    // TODO Auto-generated method stub
-
+    this.board[x][y] = 1;
   }
 
   @Override
   public void setDead(int x, int y) {
-    // TODO Auto-generated method stub
-
+    this.board[x][y] = 0;
   }
 
   @Override
-  public boolean isAlive(int x, int y) {
-    // TODO Auto-generated method stub
-    return false;
+  public int isAlive(int x, int y) {
```

Comment:
hier Verändern sie eine Methode des zu implementieren Interfaces: Das sollten Sie lieber nicht tun!
In der Praxis ist es häufig so, dass dieses Interface bereits im UI-Team verwendet wird und da verlässt man sich darauf, dass Sie ILife wirklich so implementieren, wie vereinbart!